components/NewsGrid.vue

```diff
@@ -0,0 +1,45 @@
+
+<template>
+	<div class="news-grid">
+		<div v-for="row in groupedList" :key="row.id" class="row">
+			<div v-for="x in row" :key="x.id" class="api-content">
+				<NewsItem
+					v-if="row.length>1"
+					:picture="x.picture"
+					:title="x.title"
+					:date="x.date"
+					:slug="x.slug"
+				/>
+				<NewsItem
+					v-else
+					:picture="x.picture"
+					:title="x.title"
+					:date="x.date"
+					:slug="x.slug"
+					style="width: 50%"
+				/>
+			</div>
+		</div>
+	</div>
+</template>
+
+<script>
+import _ from "lodash"
+import NewsItem from "~/components/NewsItem"
+export default {
+	components: {
+		NewsItem
+	},
+	props: {
+		postList: {
+			type: String,
+			default: () => []
```

Comment:
Does this have to be a function, or can we just set the default to `[]`?

Reply:
It has to be a function or else it throws a warning

Reply:
@TeddTech can you find an explanation of this in the documentation? That would be helpful.

Reply:
This is the error that is thrown when the default value is not in function form:

```  27:4  error  Type of the default value for 'postList' prop must be a function  vue/require-valid-default-prop```

Reply:
Actually this is interesting, I think the problem here, why you are getting the error you just mentioned and why are seeing those errors on the news-and-views page, is that you set the type to `String`. Here the default is actually an array. It probably doesn't complain about the function because it is not evaluating the function when the linting happens. However, when it runs we get the error because the array is being returned.

Reply:
@greatislander here is an explanation I found: 

> There will be a bug. because every instance will share the same thing when not passing the target property to this component.

https://github.com/vuejs/vue/issues/1032

Reply:
Could you please leave a comment in the code about this, with a link the to the issue you reference.

Reply:
@TeddTech Ideally reference the Vue docs? There should be something in there about this.

Reply:
https://vuejs.org/v2/guide/components-props.html#Prop-Validation